app/api/hooks/useApi.js

```diff
@@ -0,0 +1,24 @@
+import { useState } from "react";
+
+import { AsyncStatus } from "../../constants/enums";
+
+export default useApi = (apiFunc) => {
+  const [data, setData] = useState([]);
+  const [apiStatus, setApiStatus] = useState(AsyncStatus.Idle);
+
+  const request = async (...args) => {
+    setApiStatus(AsyncStatus.Loading);
+    const response = await apiFunc(...args);
```

Comment:
Let’s use promises for this.

Reply:
It is okay to address in next pr

Reply:
Ok, will update that in next PR

Reply:
I tried implementing the promise here just now but was facing some errors and the code started looking a little ugly. I would suggest we use async-await along with try-catch to handle all scenarios. With async-await, the code looks more maintainable and organized as compared to promise. I would like to hear your concerns about async-await. Let's discuss this during our sync up.